Sources/Starknet/Providers/StarknetProvider/JsonRpcResponse.swift

```diff
@@ -1,17 +1,29 @@
 import Foundation
 
-struct JsonRpcErrorData: Decodable {
-    let revertError: String
-
-    enum CodingKeys: String, CodingKey {
-        case revertError = "revert_error"
-    }
-}
-
 struct JsonRpcError: Decodable {
     let code: Int
     let message: String
-    let data: JsonRpcErrorData?
+    let data: String?
```

Comment:
I decided to go with this approach for now (converting any raw json to string), unless you think we should opt for fully matching errors in the spec.
In the latest spec, we have `CONTRACT_ERROR`, `TRANSACTION_EXECUTION_ERROR`  with specific `data` structure. 
  
I currently see three approaches:
1. Single class, `data` is always String? (we convert every raw json to it); Single exception
2. Multiple classes (not only those with specific `data` structure, but one for every code); A class for `InternalRpcError`; Multiple exceptions
3. Multiple classes (only those with specific `data` structure); A class for `StandardRpcError`, which handles both internal and all other errors; Multiple exceptions

For now, I decided to go with the first one.

My reasoning is:
1. We would still need to handle internal errors which would convert any json to String, so I'm not sure adding specific errors is even worth it in that context.
2. If we add types with specific fields in `data`, i.e. `ContractError`, `TransactionExecutionError`, it would then makes sense to add all the errors and all the error codes. Previously, we tried to avoid that.

Should we opt for a different approach?

Reply:
current approach seems reasonable enough as data shoud provide enough info to determine what went wrong, right?